tests/pytests/pkg/downgrade/test_salt_downgrade.py

```diff
@@ -36,17 +36,25 @@ def test_salt_downgrade(salt_call_cli, install_salt):
     assert "Authentication information could" in use_lib.stderr
 
     # Verify there is a running minion by getting its PID
+    salt_name = "salt"
     if platform.is_windows():
         process_name = "salt-minion.exe"
     else:
         process_name = "salt-minion"
-    old_pid = None
+
+    old_pid = []
+
+    # psutil process name only returning first part of the command '/opt/saltstack/'
+    # need to check all of command line for salt-minion
+    # ['/opt/saltstack/salt/bin/python3.10 /usr/bin/salt-minion MultiMinionProcessManager MinionProcessManager']
+    # and psutil is only returning the salt-minion once
     for proc in psutil.process_iter():
-        if process_name in proc.name():
-            if psutil.Process(proc.ppid()).name() != process_name:
-                old_pid = proc.pid
-                break
-    assert old_pid is not None
+        if salt_name in proc.name():
+            cmdl_strg = " ".join(str(element) for element in proc.cmdline())
+            if process_name in cmdl_strg:
```

Comment:
Do we need to build a string to check if `salt_name` is in it?
Isn't `salt_name` one of the list items?

ie, `["the", "command", salt_name, "etc"]` and if this is the case, then `salt_name in proc.cmdline()` is enough...

Reply:
proc.name() is only returning a limited '/opt/saltstack/salt', hence checking for 'salt' as a quick check, before doing more work to examine the command line.

It is only a test, and could examine all command lines, but why do extra work when you can quickly check for 'salt' in the proc.name().  Using 'salt' in case we ever change from 'saltstack' which had been banded about in the past.